File: crates/fieldloop-py/examples/quickstart.py
import uuid

import fieldloop


cap = fieldloop.Capture("tenant-a", "robot-1", capacity=1024)

ctx = cap.register_context(
    policy_version="pick@v1+abc123abc123",
    model_hash="sha256:model-hash",
    embodiment="warehouse_pick_arm",
    task_id="pick-can",
)

episode_id = str(uuid.uuid4())
rollout_id = cap.log_step(
    episode_id,
    step_index=0,
    ctx=ctx,
    inference_us=180,
)

rows = cap.drain()

print("rollout_id:", rollout_id)
print("drained:", len(rows))
print("dropped:", cap.dropped())

File: crates/fieldloop-capture/examples/quickstart.rs
use fieldloop_capture::Capture;
use fieldloop_types::{EpisodeId, PolicyVersion, RobotId, RobotIdentity, TenantId};

fn main() {
    let robot = RobotIdentity::new(TenantId::new("tenant-a"), RobotId::new("robot-1"));
    let (capture, drain) = Capture::new(robot, 1024);

    let ctx = capture.register_context(
        PolicyVersion::new("pick@v1+abc123abc123"),
        "sha256:model-hash",
        "warehouse_pick_arm",
        "pick-can",
    );

    let rollout_id = capture.log_step(EpisodeId::new(), 0, ctx, 180);
    let rollouts = drain.drain_available();

    println!("rollout_id = {rollout_id}");
    println!("drained = {}", rollouts.len());
    println!("dropped = {}", capture.dropped());
}
